h2o-algos/src/main/java/hex/coxph/CoxPH.java

```diff
@@ -547,15 +556,59 @@ protected void calcCumhaz_0(CoxPHModel model, final CoxPHTask coxMR) {
           throw new IllegalArgumentException("_ties method must be either efron or breslow");
       }
 
+      double[] totalRisks = coxMR.totalRisk.clone();
+      double[] sumHaz = new double[totalRisks.length];
+
+      for (int i = sumHaz.length - 1; i >= 0; i--) {
+        sumHaz[i] = 0d;
+      }
+
+      for (int t = 0; t < coxMR._time.length; ++t) {
+        o._baseline_hazard_matrix.set(t,0, coxMR._time[t]);
+        o._baseline_survival_matrix.set(t,0, coxMR._time[t]);
+
+        for (int strata = 0; strata < num_strata; strata++) {
+          double weightEvent = coxMR.sumWeightInStrata[t + coxMR._time.length * strata];
+          double sumRiskEvent = coxMR.sumRiskAllEvents[t + coxMR._time.length * strata];
+
+          double eventRisk = weightEvent / totalRisks[strata];
+
+          totalRisks[strata] -= sumRiskEvent;
+          
+          sumHaz[strata] += eventRisk;
+          o._baseline_hazard_matrix.set(t, strata + 1, eventRisk);
+          o._baseline_survival_matrix.set(t, strata + 1, Math.exp(-sumHaz[strata]));
+        }
+      }
+      
       for (int t = 1; t < o._cumhaz_0.length; ++t) {
         o._cumhaz_0[t]     = o._cumhaz_0[t - 1]     + o._cumhaz_0[t];
         o._var_cumhaz_1[t] = o._var_cumhaz_1[t - 1] + o._var_cumhaz_1[t];
         for (int j = 0; j < n_coef; ++j)
           o._var_cumhaz_2_matrix.set(t, j, o._var_cumhaz_2_matrix.get(t - 1, j) + o._var_cumhaz_2_matrix.get(t, j));
       }
 
-      // install var_cumhaz Frame into DKV
+      // install MatricFrames into DKV
       o._var_cumhaz_2_matrix.toFrame(o._var_cumhaz_2);
+      final Frame baselineHazardAsFrame = o._baseline_hazard_matrix.toFrame(o._baseline_hazard);
+      final Frame baselineSurvivalAsFrame = o._baseline_survival_matrix.toFrame(o._baseline_survival);
+
+      if (null == o._strataMap || 0 == o._strataMap.size()) {
+        baselineHazardAsFrame.setNames(new String[]{"t", "baseline hazard"});
+        baselineSurvivalAsFrame.setNames(new String[]{"t", "baseline survival"});
+      } else {
+        List<String> names = o._strataMap.entrySet().stream()
+                .sorted(Comparator.comparingInt(e -> e.getValue()._val))
+                .map(Map.Entry::getKey)
+                .map(i -> i._gs)
+                .map(a -> Arrays.stream(a).mapToInt(d -> (int)d).toArray())
+                .map(Arrays::toString)
+                .map(s -> s.replace(']',')').replace('[', '('))
```

Comment:
Creative but it would look nicer with a utility method that produces output that you expect (ArrayUtils)